agent/agentssh/agentssh.go

```diff
@@ -115,11 +115,15 @@ func NewServer(ctx context.Context, logger slog.Logger, prometheusRegistry *prom
 	// Clients' should ignore the host key when connecting.
 	// The agent needs to authenticate with coderd to SSH,
 	// so SSH authentication doesn't improve security.
-	randomHostKey, err := rsa.GenerateKey(rand.Reader, 2048)
+
+	// Create a deterministic random source
+	// nolint: gosec
+	deterministicRand := rand.New(rand.NewSource(42))
+	coderHostKey, err := rsa.GenerateKey(deterministicRand, 2048)
```

Comment:
Could we base this on workspace and agent name? I'm wondering if there won't be conflicts between multiple workspaces having the same host key?

Reply:
Good point. I don't think there should be any issues, but I'll look into getting the names in as seed values.

Reply:
So, we need at least one host key to start the SSH server; thus, the hardcoded `42` seed. Later, once we receive the manifest, we use the workspace ID as a seed and derive a host key, replacing the host key derived from the `42` seed.

I am unsure if deriving the key from the workspace and agent names/UUIDs makes sense, as we need the host key fixed on a per-workspace basis. If we include the agent as a seed, connections using Coder Desktop's VPN would be presented with a different host key for the same workspace, thus still refusing to connect, assuming it's an impersonation.

Reply:
The agent UUID will change every start, but the name should be stable. It is, however, possible to rename a workspace, but allowing that is a separate option that needs to be enabled. One option is `workspaceUUID.agentName`, which is as stable as it gets.